lib/rules/keyframe-selector-notation/index.mjs

```diff
@@ -72,27 +75,18 @@ const rule = (primary) => {
 					: [];
 
 			atRuleKeyframes.walkRules((keyframeRule) => {
-				transformSelector(result, keyframeRule, (selectors) => {
-					let first = true;
+				const selectors = parseSelector(getRuleSelector(keyframeRule), result, keyframeRule);
 
-					selectors.walkTags((selectorTag) => {
-						if (first && namedTimelineRangeKeywords.has(selectorTag.value)) {
-							return false;
-						}
+				if (!selectors) return;
 
-						first = false;
+				selectors.each((selector) => {
+					const keyframeSelector = parseKeyframeSelector(selector);
 
-						checkSelector(selectorTag, optionFuncs[primary]);
-					});
-				});
+					if (!keyframeSelector) return;
+
+					const { expFunc, fixFunc } = optionFuncs[primary];
```

Comment:
[suggestion] Small refactoring ideas:

- It seems better to move `const { expFunc, fixFunc } = optionFuncs[primary];` before `root.walkAtRules()` call because `optionFuncs[primary]` is used only once.
- `Object.freeze()` for `optionFuncs` seems a bit overkill because it's clear not to be mutated.

E.g.

```js
const optionFuncs = {
	// as-is...
};

const { expFunc, fixFunc } = optionFuncs[primary];

root.walkAtRules(/^(-(o|moz|ms|webkit)-)?keyframes$/i, (atRuleKeyframes) => {
	// ...
```

Reply:
Yes, makes sense! good catch 🙇